Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebExtension.h"
+
+#include "Logging.h"
+
+#include <wtf/text/Base64.h>
+
+namespace WebKit {
+
+static String const generatedBackgroundPageFilename = "_generated_background_page.html"_s;
+static String const generatedBackgroundServiceWorkerFilename = "_generated_service_worker.js"_s;
+
+std::span<const uint8_t> WebExtension::resourceDataForPath(String path, RefPtr<API::Error> *outError, CacheResult cacheResult, SuppressNotFoundErrors suppressErrors)
+{
+    ASSERT(path);
+
+    if (outError)
+        *outError = nullptr;
+
+    if (path.startsWith("data:"_s)) {
+        if (auto base64Position = path.find(";base64,"_s); base64Position != notFound) {
+            auto base64String = base64DecodeToString(path.substring(base64Position));
+            return base64String.span8();
```

Comment:
I know I recommended it, but I believe this `.span8()` usage isn't actually safe, so the previous code is probably fine.

I believe if you create a `String`, `.span8()` to it is effectively a pointer, so its safe to use in that scope but returning it means `String` could be freed.